modules/images/webp-uploads/load.php

```diff
@@ -35,5 +35,186 @@ function webp_uploads_filter_image_editor_output_format( $output_format, $filena
+			}
+
+			// TODO: Store the file size of the created image.
+			// $image['filesize'] = filesize( $image['path'] );
+			// Remove duplicated properties from the size image.
+			unset( $image['path'], $image['height'], $image['width'], $image['mime-type'] );
+
+			$sources[ $mime ] = $image;
+		}
+
+		$current_size['sources'] = $sources;
+		$sizes[ $size ]          = $current_size;
+	}
+
+	$metadata['sizes'] = $sizes;
+
+	return $metadata;
+}
+
+/**
+ * List all the available image sizes as well with the properties for each, the properties included
+ * are: width, height and crop. The logic behind this function tries to use the defined sizes by
+ * WordPress if the values is not found on the options it would fallback to the user defined
+ * sizes. Each property uses the default for each property `null` for height and width and `false`
+ * for crop the same way as `make_subsize` the editor image.
+ *
+ * @since n.e.x.t
+ *
+ * @see   WP_Image_Editor_Imagick::make_subsize()
+ * @see   WP_Image_Editor_GD::make_subsize()
+ * @return array An array with the details of all available image sizes: width, height and crop.
+ */
+function webp_uploads_get_image_sizes() {
+	$wp_image_sizes = wp_get_additional_image_sizes();
+	$sizes          = array();
+
+	// Create the full array with sizes and crop info.
+	foreach ( get_intermediate_image_sizes() as $size ) {
+		// Set the default values similar to `make_subsize`.
+		$width  = null;
+		$height = null;
+		$crop   = false;
+
+		if ( array_key_exists( $size, $wp_image_sizes ) ) {
+			if ( array_key_exists( 'width', $wp_image_sizes[ $size ] ) ) {
+				$width = (int) $wp_image_sizes[ $size ]['width'];
+			}
+			if ( array_key_exists( 'height', $wp_image_sizes[ $size ] ) ) {
+				$height = (int) $wp_image_sizes[ $size ]['height'];
+			}
+			if ( array_key_exists( 'crop', $wp_image_sizes[ $size ] ) ) {
+				$crop = (bool) $wp_image_sizes[ $size ]['crop'];
+			}
+		}
+
+		/**
+		 * Get the values from the option if is not present on the options' fallback to the
+		 * defined image sizes instead due if the option is not present it means is a custom image size.
+		 */
+		$sizes[ $size ] = array(
+			'width'  => get_option( $size . '_size_w', $width ),
+			'height' => get_option( $size . '_size_h', $height ),
+			'crop'   => (bool) get_option( $size . '_crop', $crop ),
+		);
+	}
+
+	return $sizes;
+}
+
+/**
+ * Return an array with the list of valid mime types for a sources' property.
+ *
+ * @since n.e.x.t
+ *
+ * @todo Add a filter to support more mime types.
+ *
+ * @return string[] An array of valid mime types, where the key is the mime type and the value is the extension type.
+ */
+function webp_uploads_valid_image_mime_types() {
+	return array(
+		'image/jpeg' => 'jpg',
+		'image/webp' => 'wepb',
+	);
+}
+
+add_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format', 10, 3 );
+add_filter( 'wp_generate_attachment_metadata', 'webp_uploads_create_sources_property', 10, 2 );
```

Comment:
See my comment above, let's move the `add_filter` calls right below their respective function.